### ISSUE TITLE
New messages are not routed.

### ISSUE THREAD

**Issue by sunny1978** [jcustenborder/kafka-connect-rabbitmq#3]
I have the connector up and running
all 3 kafka, zoo and schema are running too on same node.
sent a message from Pika Python command line. 
This is not received. Logs of connector doesnt progress or sent to kafka.
When I shut down connector and restart, then it receives and attempts to send it to kafka.

**Comment by jcustenborder** [comment]
This isn't much to work on. Here is how I attempted to reproduce this with Docker for Mac. 

1. Launch the docker environment `docker-compose up -d`
1. [Create the topic](http://localhost:15672/#/queues). `test` is the topic name.
1. Launch the Connector. `./bin/debug.sh`
1. Launch the console consumer. `kafka-avro-console-consumer --new-consumer --bootstrap-server kafka:9092 --property schema.registry.url=http://schema-registry:8081 --topic rabbitmq.test`
1. [Publish a message](http://localhost:15672/#/queues/%2F/test) `This is a test payload`

**Expected**

The console consumer should receive a message. 

**Actual**

The console consumer received this:

```json
{
  "consumerTag": "amq.ctag-BD-uBxyYF61SEgsJ-XO_YQ",
  "envelope": {
    "deliveryTag": 1,
    "isRedeliver": false,
    "exchange": {
      "string": ""
    },
    "routingKey": {
      "string": "test"
    }
  },
  "basicProperties": {
    "contentType": null,
    "contentEncoding": null,
    "headers": {
      
    },
    "deliveryMode": {
      "int": 1
    },
    "priority": null,
    "correlationId": null,
    "replyTo": null,
    "expiration": null,
    "messageId": null,
    "timestamp": null,
    "type": null,
    "userId": null,
    "appId": null
  },
  "body": "This is a test payload"
}
```

The default behavior of a consumer is to start at the end of the topic. If you attach the consumer after sending message this could by why you are not seeing messages.

**Comment by sunny1978** [comment]
I took your latest and deployed and tested.
1: I sent a message from http://localhost:15672/#/queues/%2F/mydirectqueue
2: I received it on Kafka Consumer.
2.1: when I use this:
confluent-3.2.1/bin/kafka-console-consumer --bootstrap-server localhost:9092 --from-beginning --topic rabbitmq.Test
I received a big chunk of json having {"schema"..., "payload":{consumerTag, envelope, basicProperties} body}. Body is Encrypted
2.2: when I use kafka-avro-console-consumer, it exit with errors
Processed a total of 1 messages
[2017-05-17 20:49:30,802] ERROR Unknown error when running consumer:  (kafka.tools.ConsoleConsumer$:105)
org.apache.kafka.common.errors.SerializationException: Error deserializing Avro message for id -1
Caused by: org.apache.kafka.common.errors.SerializationException: Unknown magic byte!
[2017-05-17 20:49:30,802] ERROR Unknown error when running consumer:  (kafka.tools.ConsoleConsumer$:105)
org.apache.kafka.common.errors.SerializationException: Error deserializing Avro message for id -1
Caused by: org.apache.kafka.common.errors.SerializationException: Unknown magic byte!

3: What worries me is, no logs are being printed on the Connector program. Stays as silent

4: I tried to send message on the same queue but using Python program. Yesterday I was receiving messages and NOT today after latest upgrade.

**Comment by jcustenborder** [comment]
> I took your latest and deployed and tested.
> 1: I sent a message from http://localhost:15672/#/queues/%2F/mydirectqueue
> 2: I received it on Kafka Consumer.
> 2.1: when I use this:
confluent-3.2.1/bin/kafka-console-consumer --bootstrap-server localhost:9092 --from-beginning --topic rabbitmq.Test
I received a big chunk of json having {"schema"..., "payload":{consumerTag, envelope, basicProperties} body}. Body is Encrypted

OK so you are receiving data?

> 2.2: when I use kafka-avro-console-consumer, it exit with errors
Processed a total of 1 messages
[2017-05-17 20:49:30,802] ERROR Unknown error when running consumer: (kafka.tools.ConsoleConsumer$:105)
org.apache.kafka.common.errors.SerializationException: Error deserializing Avro message for id -1
Caused by: org.apache.kafka.common.errors.SerializationException: Unknown magic byte!
[2017-05-17 20:49:30,802] ERROR Unknown error when running consumer: (kafka.tools.ConsoleConsumer$:105)
org.apache.kafka.common.errors.SerializationException: Error deserializing Avro message for id -1
Caused by: org.apache.kafka.common.errors.SerializationException: Unknown magic byte!

Based on your last message I'm thinking you are using the json converter instead of the avro converter. That is why you could see anything in the console consumer. Check the worker configuration for kafka connect.

> 3: What worries me is, no logs are being printed on the Connector program. Stays as silent

Under normal operation I don't write a bunch of logs. I don't want to spam the logs with `read a message`, etc. If you want to see all of the log message enable trace logging for the connector. Add `log4j.logger.com.github.jcustenborder.kafka.connect.cdc.rabbitmq=TRACE` to `etc/kafka/connect-log4j.properties`

> 4: I tried to send message on the same queue but using Python program. Yesterday I was receiving messages and NOT today after latest upgrade.

I'm confused. In 1. you said that you could read data now you are saying that you cannot receive data. What are you using for a config? Are you checking the correct topic?

**Comment by sunny1978** [comment]
I am confusing you too much. 
Summary:
when published from RabbitMq - it works fine
When published from my python program - it is Not. So blame is on me. I will modify it. 

Thanks for Prompt responses.

**Comment by jcustenborder** [comment]
No problem. I'm happy to help.

**Comment by mariojunior** [comment]
Hi, I just tried your step-by-step and I can't get the expected behaviour as you said.

On my case:

1) the kafka-avro-console-consumer never gets new messages
(Looks like it's able to connect, the kafka container's log shows me the consumer is connected)

2) I start the kafka-connect-rabbitmq (via debug.sh)

3) I go to the RabbitMQ Management web console and publish a simple msg: {"name":"mario"} using to the queue 'test'. (yep, I've created the queue previously).

4) I can see the message is consumed and stays on 'Unacked' state.
4.1) I can see in the kafka-connect-rabbitmq's log the message has been consumed, but I get the following error described in the issue #21 


Maybe something has been changed on the kafka-side and the project needs some update?

Thanks.
